constraint-solver/src/quadratic_symbolic_expression.rs

```diff
@@ -113,6 +113,25 @@ impl<T: FieldElement, V: Ord + Clone + Hash + Eq> QuadraticSymbolicExpression<T,
         !self.is_quadratic()
     }
 
+    /// If the expression is a known number, returns it.
+    pub fn try_to_number(&self) -> Option<T> {
+        self.try_to_known()?.try_to_number()
+    }
+
+    /// If the expression is equal to `QuadraticSymbolicExpression::from_unknown_variable(v)`, returns `v`.
+    pub fn try_to_simple_unknown(&self) -> Option<V> {
+        if self.is_quadratic() || !self.constant.is_known_zero() {
+            return None;
+        }
+        let [(var, coeff)] = self.linear.iter().collect::<Vec<_>>()[..] else {
```

Comment:
This always copies the whole linear vector. I think you could use `try_into()` on the iterator or use `Itertools::exactly_one`.